File: Napakalaki.ruby/Monster.rb
#encoding: utf-8
require_relative 'Prize'
require_relative 'BadConsequence'

module NapakalakiGame

class Monster

  def initialize(nombre, nivel, premio, bc, l)
    @name = nombre
    @combatlevel = nivel
    @prize = premio
    @badconsequence = bc
    @levelChangeAgaistCultistPlayer = l
  end

  def getName
    @name
  end

  def getCombatLevel
    @combatlevel
  end

  def getLevelsGained
    @prize.getLevels
  end

  def getTreasuresGained
    @prize.getTreasures
  end

  def getBadConsequence
    @badconsequence
  end

  def getCombatLevelAgainstCultistPlayer
      @combatLevel + levelChangeAgainstCultistPlayer
  end

  def to_s
    "Nombre: #{@name} \nNivel de combate: #{@combatlevel} \nPremio: \n#{@prize.to_s}
Mal rollo: \n#{@badconsequence.to_s}\n\n"
  end

end

end

File: Napakalaki.java/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author victor
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String nombre, int level, BadConsequence bc, Prize price){
        name = nombre;
        combatLevel = level;
        prize = price;
        badConsequence = bc;
        levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String nombre, int level, BadConsequence bc, Prize price, int ic){
        name = nombre;
        combatLevel = level;
        prize = price;
        badConsequence = bc;
        levelChangeAgainstCultistPlayer = ic;
    }
    
    public String getName(){
        return name;
    }

    public int getCombatLevel(){
        return combatLevel;
    }
    
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTresures();
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
  
// EXAMEN
    public int getCombatLevelAgainstCultistPlayer () {
        return combatLevel + levelChangeAgainstCultistPlayer;
    }
// FIN EXAMEN
    
    @Override
    public String toString(){
        return "Name = " + name + "\nCombatLevel = " + Integer.toString(combatLevel)
                + "\nMonsterPrize:" + prize.toString() 
                + "\nMonsterBadConsequence:" + badConsequence.toString();
    }
    
}